keps/sig-storage/20191117-noupdatable-volumes.md

```diff
@@ -0,0 +1,177 @@
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+The most popular and the most convinient way of consuming Secrets and
+ConfigMaps by Pods is consuming it as a file. However, any update to
+a Secret or ConfigMap object is quickly (roughly within a minute)
+reflected in updates of the file mounted for *all* Pods consuming them.
+That means, that a bad update (push) of Secret and/or ConfigMap can
+very quickly break the whole appplication.
+
+The recommended approach for upgrading applications are obviously
+rolling upgrades. For Secrets/ConfigMaps this is translating to
+creating a new object and updating PodTemplate with the reference
+to it. However, that doesn't protect users from outages caused by
+accidental bad updates of existing Secrets and/or ConfigMaps.
+
+Moreover, the feature of updating Secrets/ConfigMaps for running
+Pods is also expensive from scalability/performance point of view.
+Every Kubelet has to watch (default) or periodically poll every
+unique Secret and ConfigMap referenced by any of the Pods it is
+running.
+
+## Motivation
+
+In this KEP, we are proposing introducing an ability to specify that
+particular Secrets/ConfigMaps should not be updated at runtime for a
+given Pods.
+Given there are a lot of users not really taking advantage of automatic
```

Comment:
Having immutable secrets/configmaps and/or locking down access to those objects still looks to me a better approach for preventing bad pushes by accident.